src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -67,6 +71,18 @@ public virtual void Abort()
             Task.Factory.StartNew(_abortedCallback, this);
         }
 
+        public string UserIdentifier
```

Comment:
I'm not comfortable with the Lazy evaluation of this. I think it should be set by the HubEndPoint after negotiate.

Reply:
Well, I don't really like idea of having public setter for that sort of things. Since I think that for end-user connection context should be something "immutable". Don't you think so?

Reply:
I agree with that too, make it an internal setter. We'll need to see what mocking this object looks like though.